tools/perf/brave_page_sets/brave_loading_desktop_pages.py

```diff
@@ -69,16 +65,69 @@ def AddStories(self, tags, urls, startup_delay=False):
 
         page_tags = tags[:]
 
-        if startup_delay:
-          shared_page_state_class = DelayedSharedDesktopPageState
-        else:
-          shared_page_state_class = shared_page_state.SharedDesktopPageState
+        shared_page_state_class = (DelayedSharedDesktopPageState
+                                   if self.startup_delay else
+                                   shared_page_state.SharedDesktopPageState)
 
         self.AddStory(
-            page_cycler_story.PageCyclerStory(url,
-                                              self,
-                                              shared_page_state_class,
-                                              cache_temperature=temp,
-                                              tags=page_tags,
-                                              name=page_name,
-                                              perform_final_navigation=True))
+            self._create_story(url,
+                               self,
+                               shared_page_state_class,
+                               cache_temperature=temp,
+                               tags=page_tags,
+                               name=page_name,
+                               perform_final_navigation=True))
+
+  def _create_story(self, url, story_set, shared_page_state_class,
+                    cache_temperature, tags, name, perform_final_navigation):
+    """ Create a story instance. To be overridden in derived classes. """
+    raise NotImplementedError
+
+
+class BraveLoadingDesktopStorySet(BaseBraveLoadingDesktopStorySet):
+  """ Brave version of LoadingDesktopStorySet with PageCyclerStory.
+
+    See loading_desktop.py for details.
+    """
+
+  def __init__(self, startup_delay=False):
+    super().__init__('brave_loading_desktop.json', startup_delay)
+
+  def _create_story(self, url, story_set, shared_page_state_class,
+                    cache_temperature, tags, name, perform_final_navigation):
+    return page_cycler_story.PageCyclerStory(
+        url,
+        story_set,
+        shared_page_state_class,
+        cache_temperature=cache_temperature,
+        tags=tags,
+        name=name,
+        perform_final_navigation=perform_final_navigation)
+
+
+class BraveMultitabLoadingDesktopStorySet(BaseBraveLoadingDesktopStorySet):
+  """ Brave version of LoadingDesktopStorySet with MultiTab story. """
+
+  def __init__(self, startup_delay=False):
+    super().__init__('brave_loading_desktop.json', startup_delay)
+
+  class MultiTabStory(page_cycler_story.PageCyclerStory):
+    """ A story that opens multiple tabs. """
+    TAB_COUNT = 10
+
+    def RunPageInteractions(self, action_runner):
+      tabs = action_runner.tab.browser.tabs
+      tabs[0].Navigate(self.url)
+      for _ in range(1, self.TAB_COUNT):
+        new_tab = tabs.New()
+        new_tab.action_runner.Navigate(self.url)
```

Comment:
This part looks good, it actually opens n-tabs.
The main questions are:
1) what metrics do we get
2) from all the tabs or from just one
3) how we aggregate them

Reply:
The main metric we are interested in is the total memory usage (private_footprint)

Reply:
Yes, I'll handle it using system_health.memory*, which will ease our implementation for the requirement.